### PR TITLE
Some fixes & improvements

### PR DESCRIPTION
**Pull request by mlyashenko**
Hey, there is an issue i described here https://github.com/mlyashenko/jaxb2ruby/issues/1
and there are several fixes for that.

### CODE REVIEW

**Comment by sshaw** [comment]
Hi @mlyashenko, thanks for the request. Would it be possible to provide the schema that caused the errors in mlyashenko/jaxb2ruby#1?

File: lib/templates/roxml.erb [hunk 1]
```diff
@@ -16,27 +16,27 @@ class <%= @class.basename %> <%= "< #{@class.superclass}" if @class.superclass %
...
+  xml_accessor <%= accessor_name(attr) %>, :from => "@<%= attr.name %>", :required => <%= attr.required? %>
+    <%- else -%>
+  xml_accessor <%= accessor_name(attr) %>, :as => <%= type_name(attr) %>, :from => "@<%= attr.name %>", :required => <%= attr.required? %>
+    <%- end -%>
+  <%- end -%>
 end
 
-<%= @class.outter_class.map { "end" }.join " " %>
-<%= @class.module.map { "end" }.join " " %>
+<%= @class.outter_class.map { "end" }.join "; " %>
```
Comment (sshaw) [inline lib/templates/roxml.erb]: Why add the `";"`?

File: bin/jaxb2ruby [hunk 2]
```diff
@@ -81,5 +81,5 @@ begin
     File.open(File.join(outdir, klass.path), "w") { |io| io.puts template.build(klass) }
   end
 rescue => e
-  abort "class generation failed: #{e}"
+  abort "class generation failed: #{e} \n#{'*' * 100}\nbacktrace: \n#{ e.backtrace.join("\n") }"
```
Comment (sshaw) [inline bin/jaxb2ruby]: Not sure how I feel about this. I don't think command line programs should spew stacktraces to stderr. Really the only exceptions that should be caught here are `JAXB2Ruby::Error`s. If it's anything else then it's a bug (hence this pull request). 

Maybe a debug that printed what the program was doing would be better?

File: lib/templates/roxml.erb [hunk 3]
```diff
@@ -5,7 +5,7 @@ require "roxml"
 require "<%= mod %>"
 <%- end -%>
 
-<%= @class.module.map { |mod| "module #{mod}" }.join " " %>
+<%= @class.module.map { |mod| "module #{mod}" }.join "; " %>
```
Comment (sshaw) [inline lib/templates/roxml.erb]: Why add the ";"?

**Comment by mlyashenko** [comment]
@sshaw, of course, the schema is https://gist.githubusercontent.com/mlyashenko/aed660867bbfebf4453a/raw/aaec83a4495e6b793792267194eadec03e1b4439/OTA_VehAvailRateRQ.xsd

File: bin/jaxb2ruby (hunk 2 above)
Comment (mlyashenko) [inline bin/jaxb2ruby]: It just a fix for debugging. Surely, implementing a debug option would be better :smile:

File: lib/templates/roxml.erb (hunk 3 above)
Comment (mlyashenko) [inline lib/templates/roxml.erb]: The request was made to notify you about the issue, originally, this line was not planned to be included, the coding style I use requires the ";" here, just skip it if you did not find this right.

**Comment by sshaw** [comment]
Thanks I'll take a look. 

I'd like to see what type is causing `node.type.nil?` and try to parse it instead.

**Comment by sshaw** [comment]
@mlyashenko can you also include the schema referenced in `OTA_VehAvailRateRQ.xsd`: `OTA_VehicleCommonTypes.xsd`, and any others that it may reference. Without it one cannot parse.

**Comment by mlyashenko** [comment]
I recreated this issue with link to schema here https://github.com/sshaw/jaxb2ruby/issues/3

**Pull request closed without merging**